packages/example-contracts/contracts/cross-chain-warriors/CrossChainWarriors.sol

```diff
@@ -89,16 +93,12 @@ contract CrossChainWarriors is
         uint256 crossChainId,
         address to,
         uint256 tokenId
-    ) external {
+    ) external payable {
         if (!_isValidChainId(crossChainId)) revert InvalidDestinationChainId();
         if (!_isApprovedOrOwner(_msgSender(), tokenId)) revert InvalidTransferCaller();
 
         uint256 crossChainGas = 18000000000000000000;
-
-        {
-            bool success = _zetaToken.transferFrom(msg.sender, address(connector), crossChainGas);
-            if (!success) revert ErrorApprovingZeta();
-        }
+        _zetaConsumer.getZetaFromEth{value: msg.value}(address(this), crossChainGas);
```

Comment:
Same here with the approval.

Reply:
And I think `zetaValueAndGas` should be the amount we got, not the minimum we wanted to get.